bindinfo/handle.go

```diff
@@ -217,6 +229,46 @@ func (h *BindHandle) DropBindRecord(record *BindRecord) (err error) {
 	return err
 }
 
+// HandleDropBindRecord execute the drop bindRecord task.
+func (h *BindHandle) HandleDropBindRecord() {
+	h.dropBindRecordMap.Lock()
+	dropBindRecordMap := copyDroppedBindRecordMap(h.dropBindRecordMap.Load().(map[string]*droppedBindRecord))
+	for key, droppedBindRecord := range dropBindRecordMap {
+		if droppedBindRecord.droppedTime.IsZero() {
+			err := h.DropBindRecord(droppedBindRecord.bindRecord)
+			if err != nil {
+				logutil.Logger(context.Background()).Error("handleDropBindRecord failed", zap.Error(err))
+			}
+			droppedBindRecord.droppedTime = time.Now()
+			continue
+		}
+
+		if time.Since(droppedBindRecord.droppedTime) > 6*time.Second {
+			delete(dropBindRecordMap, key)
+		}
+	}
+	h.dropBindRecordMap.Store(dropBindRecordMap)
+	h.dropBindRecordMap.Unlock()
+}
+
+// AddDropBindRecordTask add bindRecord to dropBindRecordMap when the bindRecord need to be deleted.
+func (h *BindHandle) AddDropBindRecordTask(dropBindRecord *BindRecord) {
+	key := dropBindRecord.OriginalSQL + ":" + dropBindRecord.Db
+	if _, ok := h.dropBindRecordMap.Value.Load().(map[string]*droppedBindRecord)[key]; ok {
+		return
+	}
+	h.dropBindRecordMap.Lock()
+	if _, ok := h.dropBindRecordMap.Value.Load().(map[string]*droppedBindRecord)[key]; ok {
+		return
```

Comment:
The lock will never be unlocked if we return here.
We need to put the unlock in defer